includes/Core/Authentication/Clients/OAuth_Client.php

```diff
@@ -404,6 +416,26 @@ function ( $scope ) {
 	public function authorize_user() {
 		$code       = htmlspecialchars( $this->context->input()->filter( INPUT_GET, 'code' ) ?? '' );
 		$error_code = htmlspecialchars( $this->context->input()->filter( INPUT_GET, 'error' ) ?? '' );
+
+		// If we have a code, check if there's a stored redirect URL to prevent duplicate setups.
+		// The OAuth2 spec requires that an authorization code can only be used once.
+		// If `fetchAccessTokenWithAuthCode()` is called more than once with the same code, Google will return an error.
+		// This commonly happens when users click the final setup button multiple times or
```

Comment:
I'd suggest a minor tweak here, just so it doesn't sound like it's happening all the time :sweat_smile: 

```suggestion
		// This may happen when users click the final setup button multiple times or
```

Reply:
Updated.